### PR TITLE
Render alert fields when performing searches

### PR DESCRIPTION
**Pull request by LawnGnome**
We have all this useful information in the web UI that comes back from GraphQL, but isn't exposed in `src`. This PR wires up a common renderer to the `actions exec` and `search` commands to render any alert that comes back in a (hopefully) human readable form.

Here are a couple of examples of it in use:

image
image

If we think this is useful, I'll add a couple of tests to the alert rendering and we can get this merged.

### CODE REVIEW

**Comment by sqs** [comment]
Definitely useful!

**Comment by eseliger** [comment]
Very cool!

**Comment by LawnGnome** [comment]
OK, this is ready for review.

**Comment by sqs** [comment]
This would help with an issue facing a customer right now: https://sourcegraph.slack.com/archives/CJX299FGE/p1592345898255500.

**Comment by LawnGnome** [comment]
> Visually, though, I think this might be a bit hard to parse, since we have up to 4 colors:

Yeah, I'm not going to say the colours were picked _totally_ at random, but it's not far off. Here was my logic:

* The actual alert gets shown in red, the colour of errors.
* The description that goes along with it to describe a possible course of action is orange, to look like a lesser part of the error.
* The suggestions are in blue to highlight that you might want to copy/paste them.
* Everything else is in the default colour.

The warning is rendered separately in yellow.

I'm open to suggestions: unifying the description colour with the warning might make sense, for example, and I'm not tied to blue for the suggestions. (We could also bold them instead of changing colour.)

**Comment by mrnugget** [comment]
> I'm open to suggestions: unifying the description colour with the warning might make sense, for example, and I'm not tied to blue for the suggestions. (We could also bold them instead of changing colour.)

I think having the description in red also makes sense, since you could then visually group the messages: yellow warning because it didn't find anything, red error shows why it didn't find anything, then suggestions.

But don't feel blocked by this comment. I think the code can be merged. If you do have an idea to improve the colors though, go for it. If not, we can fix it later.

**Comment by LawnGnome** [comment]
I made two changes, and they look like this:

image

I adopted @mrnugget's suggestion of unifying the alert and description colours, and I think it looks good, so let's go with that. I also added an indent on the lines past the alert — as long as the emoji is rendered as a double-width character (which I think is true on most platforms), the alert will now appear together.

**Comment by mrnugget** [comment]
Looks good! Feel free to merge.

**Comment by mrnugget** [comment]
@LawnGnome Can you add a `CHANGELOG.md` entry?

File: cmd/src/search_alert_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			"[[search-alert-proposed-query]]xyz:abc[[nc]] - [[search-alert-proposed-description]]quux[[nc]]\n",
+			"[[search-alert-proposed-query]]def:ghi[[nc]] - [[search-alert-proposed-description]]baz[[nc]]\n",
+		}, ""))
+		if diff := cmp.Diff(expected, content); diff != "" {
+			t.Errorf("full alert content incorrect: %s", diff)
+		}
+	})
+}
+
+var subColourCodesRegex = regexp.MustCompile(`\[\[[a-zA-Z0-9-]+\]\]`)
```
[the remark this answers is not in the document]
Comment (LawnGnome) [reply cmd/src/search_alert_test.go:101]: Thanks. Sorry, I'm probably going to be sloppy on that for a while.

**Comment by LawnGnome** [comment]
> @LawnGnome Can you add a `CHANGELOG.md` entry?

Yep, doing it now.

File: cmd/src/search_alert_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (eseliger) [reply cmd/src/search_alert_test.go]: No worries at all, just wanted to guide you to that rule in case you didn't know about it! I'm not even a native speaker so expect much weirder constructions from words from my end haha.
